components/TodoInput.tsx

```diff
@@ -0,0 +1,53 @@
+import { StyleSheet } from "react-native";
+import { useState } from "react";
+import { TextInput, View,Button,Image } from "react-native";
+
+const TodoInput  = (props) => {
+    const [enteredTodoText, setEnteredTodoText] = useState('');
```

Comment:
This is very good to learn state
 However this code has performance issues. Please add console log below and you will see every key stroke it re-renders. After this please change to useRef technique to save latest text, and notice how the console log doesn't run everytime you type a letter.